codec/src/types/primitives.rs

```diff
@@ -189,6 +227,32 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_usize() {
+        let values = [0usize, 1, 42, u32::MAX as usize];
+        for value in values.iter() {
+            let encoded = value.encode();
+            assert_eq!(value.encode_size(), VarUInt(*value as u32).encode_size());
+            let decoded = usize::decode_cfg(encoded, &..).unwrap();
+            assert_eq!(*value, decoded);
+        }
+    }
+
+    #[test]
+    #[should_panic(expected = "encode_size: usize value is larger than u32")]
+    fn test_usize_encode_panic() {
+        let value: usize = usize::MAX;
```

Comment:
You may want to convert from u64::MAX here and fail with a different error (otherwise this could pass on a 32-bit machine)?